ts/main.ts

```diff
@@ -39,6 +57,87 @@ $entryForm.addEventListener('submit', (event: Event) => {
   data.nextEntryId++;
   data.entries.unshift(entryValues);
   writeData();
+  const $newEntry = renderEntry(entryValues);
+  $allEntriesUl.prepend($newEntry);
   $entryImg.setAttribute('src', '/images/placeholder-image-square.jpg');
   $entryForm.reset();
+  viewSwap('entries');
+  if (data.entries.length > 0) {
+    toggleNoEntries();
+  }
+});
+
+function renderEntry(entry: Entry): HTMLLIElement {
+  const $entryLI = document.createElement('li');
+
+  const $rowDiv = document.createElement('div');
+  $rowDiv.className = 'row';
+
+  const $imgDiv = document.createElement('div');
+  $imgDiv.className = 'column-half';
+
+  const $entryImg = document.createElement('img');
+  $entryImg.setAttribute('src', entry.photoUrl);
+  $entryImg.setAttribute('alt', 'image of' + entry.title);
+
+  const $textDiv = document.createElement('div');
+  $textDiv.className = 'column-half';
+
+  const $titleH3 = document.createElement('h3');
+  $titleH3.textContent = entry.title;
+
+  const $notesP = document.createElement('p');
+  $notesP.textContent = entry.notes;
+
+  $entryLI.appendChild($rowDiv);
+  $rowDiv.appendChild($imgDiv);
+  $imgDiv.appendChild($entryImg);
+  $rowDiv.appendChild($textDiv);
+  $textDiv.appendChild($titleH3);
+  $textDiv.appendChild($notesP);
+
+  return $entryLI;
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+  for (let i = 0; i < data.entries.length; i++) {
+    const $newEntry = renderEntry(data.entries[i]);
+    $allEntriesUl.appendChild($newEntry);
+  }
+  viewSwap(data.view);
+  if (data.entries.length > 0) {
+    toggleNoEntries();
+  }
+});
+
+function toggleNoEntries(): void {
+  if (!$noEntriesLi) throw new Error('$noEntriesLi query failed');
+  if ($noEntriesLi.matches('.hidden')) {
+    $noEntriesLi.className = '';
+  } else {
+    $noEntriesLi.className = 'hidden';
+  }
+}
```

Comment:
Instead of checking the `length` of your `data.entries` and determining to run the function, you should just have this condition _inside_ of your `toggleNoEntries` function and based on whether it is `> 0` or not you _add_ or _remove_ the `hidden` class